amethyst_renderer/src/vertex.rs

```diff
@@ -0,0 +1,74 @@
+//! Built-in vertex formats.
+
+use gfx;
+use gfx::format::Format;
+use gfx::pso::buffer::Structure;
+use gfx::traits::Pod;
+
+/// Handle to a vertex attribute.
+pub type Attribute = gfx::pso::buffer::Element<Format>;
+
+/// Trait implemented by all valid vertex formats.
+pub trait VertexFormat: Pod + Structure<Format> + Sized {
+    /// Returns a list of all attributes specified in the vertex.
+    fn attributes() -> Vec<Attribute>;
```

Comment:
`Attribute`s must be paired with names to make `<RawVertexBuffer as DataLink>::Init`
But user don't know which attribute corresponds to which field without looking into code.
It he looks into code then he can create static array to link RawVertexBuffer to attributes in shader without this method.